### PR TITLE
Fixing intermittent parse exceptions, chunk stream ended unexpectedly, NPE when using entitlement mediator under high concurrency. ESBJAVA-3231

### PR DESCRIPTION
**Pull request by omindu**

### CODE REVIEW

File: components/identity/org.wso2.carbon.identity.entitlement.proxy/src/main/org/wso2/carbon/identity/entitlement/proxy/soap/basicAuth/BasicAuthEntitlementServiceClient.java [hunk 1]
```diff
@@ -241,6 +237,10 @@ public boolean subjectCanActOnResource(String subjectType, String alias, String
     }
 
     private EntitlementServiceStub getEntitlementStub(String serverUrl) throws Exception {
+
+        if (configurationContext == null){
+            throw new Exception("Cannot initialize EntitlementServiceStub with null Axis2 configuration context.");
```
Comment (prabathabey) [inline components/identity/org.wso2.carbon.identity.entitlement.proxy/src/main/org/wso2/carbon/identity/entitlement/proxy/soap/basicAuth/BasicAuthEntitlementServiceClient.java]: Why is a generic exception thrown?

Comment (omindu) [inline components/identity/org.wso2.carbon.identity.entitlement.proxy/src/main/org/wso2/carbon/identity/entitlement/proxy/soap/basicAuth/BasicAuthEntitlementServiceClient.java]: Fixed in https://github.com/wso2/carbon-identity/pull/787/files

File: components/identity/org.wso2.carbon.identity.entitlement.proxy/src/main/org/wso2/carbon/identity/entitlement/proxy/soap/basicAuth/BasicAuthEntitlementServiceClient.java [hunk 2]
```diff
@@ -241,6 +237,10 @@ public boolean subjectCanActOnResource(String subjectType, String alias, String
     }
 
     private EntitlementServiceStub getEntitlementStub(String serverUrl) throws Exception {
+
+        if (configurationContext == null){
```
Comment (nirmal070125) [inline components/identity/org.wso2.carbon.identity.entitlement.proxy/src/main/org/wso2/carbon/identity/entitlement/proxy/soap/basicAuth/BasicAuthEntitlementServiceClient.java]: Formatting issue.

Comment (omindu) [inline components/identity/org.wso2.carbon.identity.entitlement.proxy/src/main/org/wso2/carbon/identity/entitlement/proxy/soap/basicAuth/BasicAuthEntitlementServiceClient.java]: Fixed in https://github.com/wso2/carbon-identity/pull/787/files

**Pull request merged**